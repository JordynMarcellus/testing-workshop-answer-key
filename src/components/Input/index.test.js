import React from "react";
import { Input } from "./";
import { render, fireEvent, cleanup } from "react-testing-library";

describe("Input component", () => {
  afterEach(cleanup);
  it("renders the input correctly", () => {
    const { container, debug } = render(
      <Input
        id="test-id"
        label="Input test"
        handleChange={jest.fn()}
        value={"Not an empty string"}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("handles an onChange event successfully", () => {
    const mockOnChange = jest.fn(event => {
      expect(event.target.value).toEqual("pikachu");
    });
    const { getByLabelText, debug } = render(
      <Input
        id="test-id"
        label="Input test"
        handleChange={mockOnChange}
        value={""}
      />
    );
    fireEvent.change(getByLabelText("Input test"), {
      target: { value: "pikachu" },
    });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
