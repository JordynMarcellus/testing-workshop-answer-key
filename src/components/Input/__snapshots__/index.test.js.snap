// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders the input correctly 1`] = `
<label
  for="test-id"
>
  Input test
</label>
`;
